### PR TITLE
feat(kafka): Set and send SASL extensions

### PR DESCRIPTION
**Pull request by gunnaraasen**
### Required checklist
- [x] Sample config files updated (both `/etc` folder and `NewDemoConfig` methods) (influxdb and plutonium)
- [ ] ~~openapi swagger.yml updated (if modified API) - link openapi PR~~
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### Description
Allows SASL extensions to be set via config. Also updates the `samara` dependency to use the new IBM source.

### CODE REVIEW

**Comment by gunnaraasen** [comment]
Looks like there's a bug. But I'm having a hard time deciphering the test code.

```
=== RUN   TestServer_AlertHandlers
=== RUN   TestServer_AlertHandlers/alerta-0
=== RUN   TestServer_AlertHandlers/bigpanda-1
=== RUN   TestServer_AlertHandlers/discord-2
=== RUN   TestServer_AlertHandlers/exec-3
=== RUN   TestServer_AlertHandlers/hipchat-4
=== RUN   TestServer_AlertHandlers/kafka-5
2024/08/29 17:22:59 Initializing new client
2024/08/29 17:22:59 Successfully initialized new client
2024/08/29 17:22:59 client/metadata fetching metadata for [test] from broker [::]:39117
2024/08/29 17:22:59 Connected to broker at [::]:39117 (unregistered)
2024/08/29 17:22:59 client/metadata got error from broker -1 while fetching metadata: kafka: insufficient data to decode packet, more bytes expected
2024/08/29 17:22:59 Closed connection to broker [::]:39117
2024/08/29 17:22:59 client/metadata no available broker to send metadata request to
2024/08/29 17:22:59 client/brokers resurrecting 1 dead seed brokers
2024/08/29 17:22:59 client/metadata retrying after 250ms... (2 attempts remaining)
2024/08/29 17:22:59 client/metadata fetching metadata for [test] from broker [::]:39117
2024/08/29 17:22:59 Connected to broker at [::]:39117 (unregistered)
2024/08/29 17:22:59 client/metadata got error from broker -1 while fetching metadata: kafka: insufficient data to decode packet, more bytes expected
2024/08/29 17:22:59 Closed connection to broker [::]:39117
2024/08/29 17:22:59 client/metadata no available broker to send metadata request to
2024/08/29 17:22:59 client/brokers resurrecting 1 dead seed brokers
2024/08/29 17:22:59 client/metadata retrying after 250ms... (1 attempts remaining)
2024/08/29 17:22:59 client/metadata fetching metadata for [test] from broker [::]:39117
2024/08/29 17:22:59 Connected to broker at [::]:39117 (unregistered)
2024/08/29 17:22:59 client/metadata got error from broker -1 while fetching metadata: kafka: insufficient data to decode packet, more bytes expected
2024/08/29 17:22:59 Closed connection to broker [::]:39117
2024/08/29 17:22:59 client/metadata no available broker to send metadata request to
2024/08/29 17:22:59 client/brokers resurrecting 1 dead seed brokers
2024/08/29 17:23:00 client/metadata retrying after 250ms... (0 attempts remaining)
2024/08/29 17:23:00 client/metadata fetching metadata for [test] from broker [::]:39117
2024/08/29 17:23:00 Connected to broker at [::]:39117 (unregistered)
2024/08/29 17:23:00 client/metadata got error from broker -1 while fetching metadata: kafka: insufficient data to decode packet, more bytes expected
2024/08/29 17:23:00 Closed connection to broker [::]:39117
2024/08/29 17:23:00 client/metadata no available broker to send metadata request to
2024/08/29 17:23:00 client/brokers resurrecting 1 dead seed brokers
    server_test.go:11907: kafka: unexpected kafka messages -exp/+got:
          []kafkatest.Message(
        -       {
        -               {
        -                       Topic:     "test",
        -                       Partition: 3,
        -                       Key:       "id",
        -                       Message:   `{"id":"id","message":"message","details":"details","time":"1970-`...,
        -                       Time:      s"2024-08-29 17:24:44.491478142 +0000 UTC",
        -               },
        -       },
        +       nil,
          )
--- FAIL: TestServer_AlertHandlers (130.34s)
    --- PASS: TestServer_AlertHandlers/alerta-0 (5.04s)
    --- PASS: TestServer_AlertHandlers/bigpanda-1 (5.02s)
    --- PASS: TestServer_AlertHandlers/discord-2 (5.03s)
    --- PASS: TestServer_AlertHandlers/exec-3 (5.02s)
    --- PASS: TestServer_AlertHandlers/hipchat-4 (5.04s)
    --- FAIL: TestServer_AlertHandlers/kafka-5 (105.19s)
```

File: services/kafka/service.go [hunk 1]
```diff
@@ -218,7 +218,7 @@ func configChanged(old, new Config) bool {
 		old.SSLCA != new.SSLCA ||
 		old.SSLCert != new.SSLCert ||
 		old.SSLKey != new.SSLKey ||
-		old.SASLAuth != new.SASLAuth
+		fmt.Sprint(old.SASLAuth) != fmt.Sprint(new.SASLAuth)
```
Comment (vlastahajek) [inline services/kafka/service.go:221]: Definitely not the right way of comparing complex types
```suggestion
		!cmp.Equal(old.SASLAuth, new.SASLAuth)
```

**Review by vlastahajek — CHANGES_REQUESTED** [review]
The test failed because of the upgraded Sarama package (I haven't investigated the exact problem). For simplicity, I suggest keeping the previous Sarama library version.

**Review by bednar — APPROVED** [review]
Looks good to me 🚀

Following our discussion here: https://influxdata.slack.com/archives/C5BSZ026L/p1725387374078259, it is okay to use Sarama v1.40.1.

**Review by vlastahajek — APPROVED** [review]

**Pull request merged**